### ISSUE TITLE
replacing modern.rkt implementation with readtable-based version like https://github.com/AlexKnauth/sweet-racket/blob/readtable2/sweet-exp2/modern/reader.rkt

### ISSUE THREAD

**Issue by AlexKnauth** [takikawa/sweet-racket#26]
@takikawa 
Would it make sense to replace the modern.rkt implementation with a readtable-based version like https://github.com/AlexKnauth/sweet-racket/blob/readtable2/sweet-exp2/modern/reader.rkt
?

By the way I was first thinking of this as a way to define a `sweet-exp2` as a language that would be able to take `#lang sweet-exp2 at-exp racket` or something like that, but I'm not quite there with that yet.  But even by itself, the readtable-based version simplifies a lot of things and avoids reimplementations.

**Comment by AlexKnauth** [comment]
But for some reason when I try this:
https://github.com/AlexKnauth/sweet-racket/commit/624dd677b96b99a3475edd96ce50338b81922c03
It gives me errors like this:
https://travis-ci.org/AlexKnauth/sweet-racket/jobs/58120454#L166

```
sweet-exp/tests/hash.rkt:3:0: read: bad syntax `#hash('
```

So do you have any idea why that's not working properly?

**Comment by bb010g** [comment]
Has there been any progress on this recently? I'm all for composibility and simplification, and the new code is a lot nicer to read.

**Comment by AlexKnauth** [comment]
Not really, because of the `#hash()` issue, though that will be fixed in the next release of racket.

**Comment by bb010g** [comment]
Has the #hash issue been fixed yet?

**Comment by AlexKnauth** [comment]
I think so, but I haven't tried it since then. I will work on it some more and see if I can get it to work.

If it does work, then if I remember correctly, the next problem was going to be rest-dots, as in `a . b`. I'm not sure how I should solve that yet.

**Comment by bb010g** [comment]
Bump.

**Comment by AlexKnauth** [comment]
I'm pretty busy right now, and don't have time to work on this.
